x-pack/libbeat/management/api/configuration.go

```diff
@@ -50,16 +51,40 @@ func (c *ConfigBlock) ConfigWithMeta() (*reload.ConfigWithMeta, error) {
 	}, nil
 }
 
+type configResponse struct {
+	Type string
+	Raw  map[string]interface{}
+}
+
+func (c *configResponse) UnmarshalJSON(b []byte) error {
```

Comment:
nit: as this is a "constructor/parser" json to `*configResponse` i'd prefer this to be a function like:

```
func unmarshalConfigReponse(json []byte) (*configResponse, error) {
  ...
}
```